### PR TITLE
Add support for userspace to docker build commands.

### LINKED ISSUE #18130: [Bug] Container build script mismatching permissions?
Following the [getting started docker guide](https://github.com/qmk/qmk_firmware/blob/master/docs/getting_started_docker.md), running `util/docker_build.sh planck/rev6:default` with Podman on Fedora 36 results in the following error:
```sh
make: stat: Makefile: Permission denied
make: *** No rule to make target 'planck/rev6:default'.  Stop.
```

### PR DESCRIPTION
**Pull request by tzarc**
## Description

As per title.  Both `./util/docker_build.sh` and`./util/docker_cmd.sh` should be sorted, now. Simplified `_build` to call `_cmd` so there's no duplicate implementations.

Also adds two new hidden CLI commands -- `qmk path` and `qmk userspace-path` to print out the computed values for both `qmk_firmware` and `qmk_userspace`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues

* Fixes #18130 as well.
* Fixes https://github.com/qmk/qmk_userspace/issues/24
* Fixes https://github.com/qmk/qmk_firmware/issues/23984

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/cli/path.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from milc import cli
```
Comment (zvecr) [inline lib/python/qmk/cli/path.py:1]: This probably doesnt need to exist, given we can do `qmk env QMK_FIRMWARE`.

File: util/docker_cmd.sh [hunk 2]
```diff
@@ -54,6 +63,7 @@ fi
 	$usb_args \
 	$uid_arg \
 	-w /qmk_firmware \
-	-v "$dir":/qmk_firmware \
+	-v "$qmk_firmware_dir":/qmk_firmware:z \
+	$userspace_docker_args \
```
Comment (zvecr) [inline util/docker_cmd.sh:67]:
```suggestion
	$userspace_docker_args \
	-e SKIP_GIT="$SKIP_GIT" \
	-e MAKEFLAGS="$MAKEFLAGS" \
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: util/docker_cmd.sh [hunk 3]
```diff
@@ -37,13 +38,21 @@ fi
...
 else
-    usb_args="--privileged -v /dev:/dev"
+	usb_args="--privileged -v /dev:/dev"
+fi
+
+qmk_firmware_dir=$(pwd -W 2>/dev/null) || qmk_firmware_dir=$PWD  # Use Windows path if on Windows
+qmk_userspace_dir=""
+userspace_docker_args=""
+
+if [ -n "$(qmk userspace-path)" ]; then
```
Comment (sigprof) [inline util/docker_cmd.sh:52]: Previously it had been possible to use the docker build commands without having the QMK CLI installed on the host.  Should we add some way to support such configuration even in the presence of userspace (e.g., by picking up `$QMK_USERSPACE` from the environment before trying to discover it through `qmk`)?

File: lib/python/qmk/cli/path.py (hunk 1 above)
Comment (tzarc) [reply lib/python/qmk/cli/path.py]: Done.

File: util/docker_cmd.sh (hunk 2 above)
Comment (tzarc) [reply util/docker_cmd.sh:72]: Added `$SKIP_VERSION`, too.

File: util/docker_cmd.sh (hunk 3 above)
Comment (tzarc) [reply util/docker_cmd.sh]: Done.

**Review by zvecr — APPROVED** [review]

**Pull request merged**